### PR TITLE
 Added developer documentation on how to write a tagging rule

### PR DESCRIPTION
**Pull request by joachimmetz**
Added developer documentation on how to write a tagging rule

### CODE REVIEW

File: docs/sources/developer/Developers-Guide.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ Once you've set up your development environment we recommend start simple:
 * [How to write a parser plugin](How-to-write-a-parser-plugin.md)
 * [How to write an analysis plugin](How-to-write-an-analysis-plugin.md)
 * [How to write an output module](How-to-write-an-output-module.md)
+* [How to write a tagging rule](How-to-write-a-tagging-rule.html)
```
Comment (rgayon) [inline docs/sources/developer/Developers-Guide.md:30]: The file you're adding is .md, why the changes to .html?
Is there some github magic happening, or is this because these are supposed to be hosted elsewhere?

Comment (joachimmetz) [reply docs/sources/developer/Developers-Guide.md:30]: I think the magic is here https://github.com/log2timeline/plaso/tree/master/docs, I need to dig into this a bit for some other PRs

Comment (rgayon) [reply docs/sources/developer/Developers-Guide.md:30]: it looks to me that we still want to link to .md here, and the link get changed to .html later on by some magic

File: docs/sources/user/Analysis-plugin-tagging.md [hunk 2]
```diff
@@ -2,37 +2,41 @@
...
+
 ```
 psort.py --analysis tagging --tagging-file tagging-file.txt timeline.plaso
 ```
 
 ## Also see
 
-* [Event filters](Event-Filters.md)
-* [Tagging rules](Taggging-Rules.md)
+* [Event filters](Event-Filters.html)
```
Comment (rgayon) [inline docs/sources/user/Analysis-plugin-tagging.md:41]: same here

File: docs/sources/developer/Developers-Guide.md (hunk 1 above)
Comment (joachimmetz) [reply docs/sources/developer/Developers-Guide.md:30]: ack, I'll change it to .md, in parallel I've started working on sphinx-doc CI tests to catch these issues.

File: docs/sources/user/Analysis-plugin-tagging.md (hunk 2 above)
Comment (joachimmetz) [reply docs/sources/user/Analysis-plugin-tagging.md:41]: ack

**Comment by joachimmetz** [comment]
@rgayon changes made, PTAL
